From 576760bff6b91f3ea8121842a4d56036151a0f79 Mon Sep 17 00:00:00 2001
From: "R.J.V. Bertin" <rjvbertin@gmail.com>
Date: Thu, 9 Mar 2017 17:13:53 +0100
Subject: [PATCH] don't add null QActions to the contextmenu

Those can occur when plugins like the debugger plugins are not loaded and
lead to qWarnings from QWidget::insertAction().
---
 interfaces/contextmenuextension.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git kdevplatform/interfaces/contextmenuextension.cpp kdevplatform/interfaces/contextmenuextension.cpp
index c76f22e..0ca493b 100644
--- kdevplatform/interfaces/contextmenuextension.cpp
+++ kdevplatform/interfaces/contextmenuextension.cpp
@@ -102,6 +102,9 @@ void populateMenuWithGroup(
     for (const ContextMenuExtension& extension : extensions) {
         groupActions += extension.actions(groupName);
     }
+    // remove NULL QActions, if any. Those can end up in groupActions if plugins
+    // like the debugger plugins are not loaded.
+    groupActions.removeAll(nullptr);
 
     if (groupActions.isEmpty()) {
         return;
-- 
2.10.2

